utils/node_local_test.py

```diff
@@ -0,0 +1,798 @@
+                raise Exception('Timeout starting dfuse')
+
+    def _close_files(self):
+        work_done = False
+        for fname in os.listdir('/proc/self/fd'):
+            try:
+                tfile = os.readlink(os.path.join('/proc/self/fd', fname))
+            except FileNotFoundError:
+                continue
+            if tfile.startswith(self.dir):
+                print('closing file {}'.format(tfile))
+                os.close(int(fname))
+                work_done = True
+        return work_done
+
+    def __del__(self):
+        if self._sp:
+            self.stop()
+
+    def stop(self):
+        """Stop a previously started dfuse instance"""
+        if not self._sp:
+            return
+
+        print('Stopping fuse')
+        ret = umount(self.dir)
+        if ret:
+            self._close_files()
+            umount(self.dir)
+
+        ret = self._sp.wait(timeout=20)
+        print('rc from dfuse {}'.format(ret))
+        self._sp = None
+        log_test(self.log_file)
+
+        # Finally, modify the valgrind xml file to remove the
+        # prefix to the src dir.
+        self.valgrind.convert_xml()
+
+    def wait_for_exit(self):
+        """Wait for dfuse to exit"""
+        ret = self._sp.wait()
+        print('rc from dfuse {}'.format(ret))
+        self._sp = None
+
+def get_pool_list():
+    """Return a list of valid pool names"""
+    pools = []
+
+    for fname in os.listdir('/mnt/daos'):
+        if len(fname) != 36:
+            continue
+        try:
+            uuid.UUID(fname)
+        except ValueError:
+            continue
+        pools.append(fname)
+    return pools
+
+def assert_file_size(ofd, size):
+    """Verify the file size is as expected"""
+    stat = os.fstat(ofd.fileno())
+    print('Checking file size is {} {}'.format(size, stat.st_size))
+    assert stat.st_size == size
+
+def import_daos(server, conf):
+    """Return a handle to the pydaos module"""
+
+    if sys.version_info.major < 3:
+        pydir = 'python{}.{}'.format(sys.version_info.major,
+                                     sys.version_info.minor)
+    else:
+        pydir = 'python{}'.format(sys.version_info.major)
+
+    sys.path.append(os.path.join(conf['PREFIX'],
+                                 'lib64',
+                                 pydir,
+                                 'site-packages'))
+
+    os.environ['CRT_PHY_ADDR_STR'] = 'ofi+sockets'
+    os.environ['OFI_INTERFACE'] = 'lo'
+    os.environ["DAOS_AGENT_DRPC_DIR"] = server.agent_dir
+
+    daos = __import__('pydaos')
+    return daos
+
+def run_daos_cmd(conf, cmd):
+    """Run a DAOS command
+
+    Run a command, returing what subprocess.run() would.
+
+    Enable logging, and valgrind for the command.
+    """
+    valgrind = ValgrindHelper()
+    exec_cmd = valgrind.get_cmd_prefix()
+    exec_cmd.append(os.path.join(conf['PREFIX'], 'bin', 'daos'))
+    exec_cmd.extend(cmd)
+
+    cmd_env = os.environ.copy()
+
+    cmd_env['D_LOG_MASK'] = 'DEBUG'
```

Comment:
Same as before. Any specific reason to use DEBUG? How about ERR? DEBUG log level can be enabled if something goes wrong for debugging purpose.

Reply:
Yes, a lot of the tests and reporting simply do not work unless this is set to DEBUG.